src/python_testing/matter_testing_infrastructure/generate_data_model_xmls_gni.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import jinja2
+
+# Set chip_root to be dynamically based on the script's location
+chip_root = os.path.abspath(os.path.join(os.path.dirname(__file__), "../../.."))
+
+# Directories to search for .xml and .json files relative to chip_root
+directories = [
+    os.path.join(chip_root, "data_model/1.3/clusters/"),
+    os.path.join(chip_root, "data_model/1.3/device_types/"),
+    os.path.join(chip_root, "data_model/1.4/clusters/"),
+    os.path.join(chip_root, "data_model/1.4/device_types/"),
+    os.path.join(chip_root, "data_model/master/clusters/"),
+    os.path.join(chip_root, "data_model/master/device_types/"),
+]
+
+# Template for generating the GNI file content
+GNI_TEMPLATE = """\
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+
+data_model_XMLS = [
+{% for name in file_list -%}
+"{{ name }}",
+{% endfor -%}
+]
+"""
+
+# Function to find and collect all .xml and .json files
+def find_files():
```

Comment:
find_files seems slightly vague ... what files. You could make it `find_files_with_extensions(extensions = [".xml", ".json"])` and then use `any([f.endswith(e) for e in extensions])` in your filters, however maybe finding a different name would be clearer. How about:

```suggestion
#
# Data model files that the module uses are currently all XML and JSON files.
#    - XMLs are generally data definitions
#    - json are currently only cluster_ids.json
def get_data_model_file_names():
```